celltype_annotation/analysis/cell-assign-sarcoma.Rmd

```diff
@@ -0,0 +1,481 @@
+  pheatmap::pheatmap(show_rownames = FALSE)
+```
+Here the majority of cells are getting classified as `other` with a small group of fibroblasts and myocytes. 
+
+Look at the marker genes that were used for the Myoblast celltype. 
+
+```{r}
+panglao_connective_tissue |>
+  dplyr::filter(`cell type` == "Myoblasts")
+```
+
+```{r}
+# add celltypes to processed sce 
+processed_sce$panglao_assignments <- panglao_assignments$celltype
+
+scater::plotExpression(processed_sce,
+                       features = myoblast_features,
+                       x = "panglao_assignments",
+                       point_size = 0.5,
+                       point_alpha = 0.5) +
+  facet_grid(row = vars(Feature)) + 
+  guides(x = guide_axis(angle = 90))
+```
+
+
+## Reference comparison 
+
+To understand how CellAssign is assigning each of the cell types vs. the original assignments, let's directly compare references below.
+
+
+```{r}
+# extract coldata for easy plotting
+coldata_df <- colData(processed_sce) |>
+  as.data.frame() |> 
+  dplyr::select(barcode, celltype, cellmarker_assignments, panglao_assignments) |>
+  # classify cells as tumor or normal
+  dplyr::mutate(cell_category = dplyr::if_else(stringr::str_detect(celltype, "Tumor"), "Tumor", celltype),
+                # remove sub states of myoblast/mesoderm/myocyte
+                broad_celltype = stringr::str_remove(celltype, "-A|-B|-C|-D"))
+```
+
+
+```{r}
+# tumor/normal vs. cellmarker 
+compare_refs_heatmap(original_assignment = coldata_df$cell_category,
+                     cell_assign = coldata_df$cellmarker_assignments)
+```
+
+
+```{r}
+# tumor/normal vs. panglao
+compare_refs_heatmap(original_assignment = coldata_df$cell_category,
+                     cell_assign = coldata_df$panglao_assignments)
+```
+
+```{r}
+# all celltypes vs. cell marker 
+compare_refs_heatmap(original_assignment = coldata_df$broad_celltype,
+                     cell_assign = coldata_df$cellmarker_assignments)
+```
+
+```{r}
+# all cell types vs. panglao
+compare_refs_heatmap(original_assignment = coldata_df$broad_celltype,
+                     cell_assign = coldata_df$panglao_assignments)
+```
+
+As expected, `CellAssign` seems to be struggling with the tumor cells and depending on the reference has been assigning them to `other` or `macrophage`. 
+There are a few places where the normal cells match up in their assignment, but they are also not perfectly in agreement. 
+
+## Conclusions
+
+- Looking at the two marker gene lists, PanglaoDb appears to have a larger number of marker gene lists for each cell type than CellMarker2.0.
+This is probably due to the way the marker gene list is organized.
+PanglaoDb only provides one list of potential marker genes for each cell type, regardless of tissue type. 
+- CellMarker2.0 has a larger list of tissue types and cell types and even includes some cancer tissue, while PanglaoDb does not. 
+Additionally, it provides the added advantage of grouping marker genes by ontology ID. 
```

Comment:
The fact that CellMarker uses cell ontology is probably its biggest benefit at this stage. PanglaoDb not using it is annoying, but maybe not awful given that we don't really trust anything at this point.